src/main/java/com/loadero/http/TestTypeAdapter.java

```diff
@@ -0,0 +1,107 @@
+    private final String INCREMENT_STRATEGY = "increment_strategy";
+    private final String MODE = "mode";
+    private final String GROUP_COUNT = "group_count";
+    private final String PARTICIPANT_COUNT = "participant_count";
+
+    @Override
+    public Test read(JsonReader jsonReader) throws IOException {
+        TestParamsBuilder paramsBuilder = TestParams.builder();
+        jsonReader.beginObject();
+
+        while (jsonReader.hasNext()) {
+            String key = jsonReader.nextName();
+
+            switch (key) {
+                case ID:
+                    paramsBuilder.withId(jsonReader.nextInt());
+                    break;
+                case PROJECT_ID:
+                    paramsBuilder.withProjectId(jsonReader.nextInt());
+                    break;
+                case NAME:
+                    paramsBuilder.withName(jsonReader.nextString());
+                    break;
+                case CREATED:
+                    paramsBuilder.withCreated(jsonReader.nextString());
+                    break;
+                case UPDATED:
+                    paramsBuilder.withUpdated(jsonReader.nextString());
+                    break;
+                case SCRIPT_FILE_ID:
+                    paramsBuilder.withScriptFileId(jsonReader.nextInt());
+                    break;
+                case START_INTERVAL:
+                    paramsBuilder.withStartInterval(Duration.ofSeconds(jsonReader.nextInt()));
+                    break;
+                case PARTICIPANT_TIMEOUT:
+                    paramsBuilder.withParticipantTimeout(Duration.ofSeconds(jsonReader.nextInt()));
+                    break;
+                case MODE:
+                    paramsBuilder.withMode(TestMode.getConstant(jsonReader.nextString()));
+                    break;
+                case INCREMENT_STRATEGY:
+                    paramsBuilder.withIncrementStrategy(
+                        IncrementStrategy.getConstant(jsonReader.nextString()));
+                    break;
+                case GROUP_COUNT:
+                    paramsBuilder.withGroupCount(jsonReader.nextInt());
+                    break;
+                case PARTICIPANT_COUNT:
+                    paramsBuilder.withParticipantCount(jsonReader.nextInt());
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        jsonReader.endObject();
+        TestParams params = paramsBuilder.build();
+
+        return new Test(params);
+    }
+
+    @Override
+    public void write(JsonWriter jsonWriter, Test test) throws IOException {
+        if (Objects.isNull(test)) {
+            jsonWriter.nullValue();
+        } else {
+            jsonWriter.beginObject();
+
+            jsonWriter.name("created").value(test.getCreated());
+            jsonWriter.name("updated").value(test.getUpdated());
+            jsonWriter.name("id").value(test.getId());
+            jsonWriter.name("project_id").value(test.getProjectId());
```

Comment:
these 4 values cannot be sent to Loadero API as they must be empty fields (readonly)